src/player.cpp

```diff
@@ -4761,6 +4761,17 @@ dealt_damage_instance player::deal_damage(Creature* source, body_part bp, const
                                 source->disp_name().c_str());
             }
         }
+
+        // Melee damage causing lacerations through filthy clothing may result in infection
+        if( d.type_damage( DT_CUT ) || d.type_damage( DT_STAB ) ) {
+            bool filthy = std::any_of( worn.begin(), worn.end(), [bp]( const item& e ) {
+                return e.covers( bp ) && e.is_filthy();
+            } );
+            if( filthy && one_in( 2 ) ) {
```

Comment:
It's not enough. Companions will complain with hardcoded: `"The bite wound on my %s looks bad."`. Also, IMHO not only the description should be changed, but the name itself everywhere in the code.